lib/Dialect/FIRRTL/FIRRTLOps.cpp

```diff
@@ -2399,6 +2419,26 @@ LogicalResult RefDefineOp::verify() {
   return success();
 }
 
+static ParseResult parseConstDecay(OpAsmParser &parser, Type &destType,
+                                   Type &srcType) {
+  if (parser.parseType(destType))
+    return failure();
+  if (succeeded(parser.parseOptionalComma())) {
+    if (parser.parseType(srcType))
+      return failure();
+  } else {
+    srcType = destType;
+  }
+  return success();
+}
+
+static void printConstDecay(OpAsmPrinter &p, Operation *op, Type destType,
+                            Type srcType) {
+  p << destType;
+  if (srcType != destType)
+    p << ", " << srcType;
```

Comment:
Does `operator<<` do the best/most appropriate thing here?  It sure might, but since already crushing it with the optional type printing/parsing, thought I'd ask.  re:qualified/dialect prefix/etc.

Reply:
Hm, in practice it seems to work. I'll look more into this, maybe the `qualified` is irrelevant here

Reply:
Sorry, I was just curious since I couldn't immediately tell.  It probably is fine, but indeed unsure if qualified is important now that you mention it.  Handle as you see fit :+1: .

Reply:
I hunted down the MLIR code that deals with `qualified`. Based on that code, the usage here is meaningless since I'm calling `operator<<`, which calls `printType` which prints the whole type. 

If `qualified` is used in a non-custom context, MLIR generates a printer that calls `printType`. If `qualified`  isn't used, a `printStrippedAttrOrType` call is generated.